lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java

```diff
@@ -0,0 +1,768 @@
+
+                                JsonNode credentialSubject =
+                                        objectMapper
+                                                .readTree(credential.getJWTClaimsSet().toString())
+                                                .get(VC)
+                                                .get(CREDENTIAL_SUBJECT);
+
+                                JsonNode nameParts =
+                                        credentialSubject.get(NAME).get(0).get(NAME_PARTS);
+                                JsonNode birthDateNode = credentialSubject.get(BIRTH_DATE).get(0);
+                                JsonNode passportNode = credentialSubject.get(PASSPORT).get(0);
+
+                                assertEquals("GivenName", nameParts.get(0).get(NAME_TYPE).asText());
+                                assertEquals(
+                                        "FamilyName", nameParts.get(1).get(NAME_TYPE).asText());
+
+                                assertEquals("Kenneth", nameParts.get(0).get(VALUE).asText());
+                                assertEquals("Decerqueira", nameParts.get(1).get(VALUE).asText());
+
+                                assertEquals("2030-01-01", passportNode.get(EXPIRY_DATE).asText());
+                                assertEquals(
+                                        "321654987", passportNode.get(DOCUMENT_NUMBER).asText());
+
+                                assertEquals("1965-07-08", birthDateNode.get(VALUE).asText());
+
+                            } catch (Exception ignored) {
+                            }
+                        });
+            } catch (Exception ignored) {
+            }
+        }
+    }
+
+    @NotNull
+    private static CredentialIssuerConfig getCredentialIssuerConfig(MockServer mockServer)
+            throws URISyntaxException {
+        return new CredentialIssuerConfig(
+                new URI("http://localhost:" + mockServer.getPort() + "/token"),
+                new URI("http://localhost:" + mockServer.getPort() + "/credential"),
+                new URI("http://localhost:" + mockServer.getPort() + "/authorize"),
+                IPV_CORE_CLIENT_ID,
+                CRI_SIGNING_PRIVATE_KEY_JWK,
+                CRI_RSA_ENCRYPTION_PUBLIC_JWK,
+                "dummyF2fComponentId",
+                URI.create(
+                        "https://identity.staging.account.gov.uk/credential-issuer/callback?id=f2f"),
+                true,
+                false);
+    }
+
+    private String createSuccessTestEvent(String jwtBody, String jwtSignature)
+            throws JsonProcessingException {
+        final CriResponseMessageDto criResponseMessageDto =
+                new CriResponseMessageDto(
+                        "f2f",
+                        TEST_USER,
+                        TEST_OAUTH_STATE,
+                        List.of(new JwtTestHelper(VALID_VC_HEADER, jwtBody, jwtSignature).build()),
```

Comment:
I suspect that this will break/be impossible for the provider side test to match. JWT signatures have a random element so the chances of the provider producing the same signature as this are extremely low.
You may need to do something similar to what we have in `PactJwtIgnoreSignatureBody` so that the generated PACT file contains a regex for the message that ignores the signature.

Reply:
Will this not remain the same because I am reusing the same signature every time?